form/src/main/java/org/dhis2/form/ui/provider/inputfield/FieldProvider.kt

```diff
@@ -102,374 +103,439 @@ fun FieldProvider(
-                    inputStyle = inputStyle,
-                    fieldUiModel = fieldUiModel,
-                    intentHandler = intentHandler,
-                    uiEventHandler = uiEventHandler,
-                    focusManager = focusManager,
-                    onNextClicked = onNextClicked,
-
-                )
-            }
+        ValueType.TRUE_ONLY -> {
+            when (fieldUiModel.renderingType) {
+                UiRenderType.TOGGLE -> {
+                    ProvideYesOnlySwitchInput(
+                        modifier = modifier,
+                        inputStyle = inputStyle,
+                        fieldUiModel = fieldUiModel,
+                        intentHandler = intentHandler,
+                    )
+                }
 
-            ValueType.DATE,
-            ValueType.DATETIME,
-            ValueType.TIME,
-            -> {
-                ProvideInputDate(
-                    modifier = modifierWithFocus,
-                    inputStyle = inputStyle,
-                    fieldUiModel = fieldUiModel,
-                    intentHandler = intentHandler,
-                    onNextClicked = onNextClicked,
-                )
+                else -> {
+                    ProvideYesOnlyCheckBoxInput(
+                        modifier = modifier,
+                        inputStyle = inputStyle,
+                        fieldUiModel = fieldUiModel,
+                        intentHandler = intentHandler,
+                        focusRequester = focusRequester,
+                    )
+                }
             }
+        }
 
-            ValueType.IMAGE -> {
-                when (fieldUiModel.renderingType) {
-                    UiRenderType.CANVAS -> {
-                        ProvideInputSignature(
-                            modifier = modifierWithFocus,
-                            fieldUiModel = fieldUiModel,
-                        )
-                    }
-
-                    else -> {
-                        ProvideInputImage(
-                            modifier = modifierWithFocus,
-                            fieldUiModel = fieldUiModel,
-                            intentHandler = intentHandler,
-                            uiEventHandler = uiEventHandler,
-                            resources = resources,
-                        )
-                    }
+        ValueType.PHONE_NUMBER -> {
+            ProvideInputPhoneNumber(
+                modifier = modifier,
+                inputStyle = inputStyle,
+                fieldUiModel = fieldUiModel,
+                intentHandler = intentHandler,
+                uiEventHandler = uiEventHandler,
+                focusManager = focusManager,
+                onNextClicked = onNextClicked,
+
+            )
+        }
+
+        ValueType.DATE,
+        ValueType.DATETIME,
+        ValueType.TIME,
+        -> {
+            when (fieldUiModel.periodSelector) {
+                null -> {
+                    ProvideInputDate(
```

Comment:
have you taken into account here that, if period type is daily, we will should  show the input Date?

Reply:
This is a great observation!, I tried to don't expose this check in the ui and what I did is to process it in EventRepository:
`private fun getPeriodSelector(): PeriodSelector? {
        return programStage?.periodType()?.let { periodType ->
            if (periodType != PeriodType.Daily) {
                PeriodSelector(
                    type = periodType,
                    minDate = getPeriodMinDate(periodType),
                    maxDate = dateUtils.today,
                )
            } else {
                null
            }
        }
    }`
So when I am going to generate the DateModel I only setUp Period model if meet requirements